_posts/2020-01-27-hhvm-4.42.markdown

```diff
@@ -0,0 +1,57 @@
+---
+title: "HHVM 4.42"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.42 is released! This release marks the end of support for 4.35;
+HHVM 4.36&ndash;4.41 remain supported, as do the 4.8 and 4.32 LTS releases.
+
+# Highlights
+
+- Bugs in the typechecker's AST handling now result in error `4361`, instead of
+  an exception. This error code should never be suppressed - please [file a bug
+  report](https://github.com/facebook/hhvm/issues/new/choose) if you see this
+  error code.
+- Fix escaping of variables autocomplete snippets - e.g. `Foo::bar|` can
+  now autocomplete to `Foo::bar($id)`.
+- Added `HHVM_VERSION_MAJOR`, `HHVM_VERSION_MINOR`, and `HHVM_VERSION_PATCH`
+  constants; these should be used instead of parsing the `HHVM_VERSION` or
+  `HHVM_VERSION_ID` constants. `HHVM_VERSION_ID` is recommended for direct
+  comparisons, e.g. with `>=`.
+- Added `\HH\enumname<T>` type; this is similar to `classname<T>`, for Hack
+  enums.
```

Comment:
I announced this one last week

Reply:
Ah; last week it was in the HHIs, but not in HHVM